bin.js

```diff
@@ -27,23 +27,24 @@ const packageDetails = require(path.join(__dirname, "package.json"));
       )
       .parse(process.argv);
 
-    const filename = program.args[0];
-
-    if (!filename) {
+    if ( 0 == program.args.length ){
       console.log("Usage: pin-github-action /path/to/workflow.yml");
       process.exit(1);
     }
 
     let allowed = program.opts().allow;
     allowed = (allowed || "").split(",").filter((r) => r);
     let ignoreShas = program.opts().ignoreShas;
-
-    const input = fs.readFileSync(filename).toString();
-
     let allowEmpty = program.opts().allowEmpty;
-    const output = await run(input, allowed, ignoreShas, allowEmpty, debug);
 
-    fs.writeFileSync(filename, output.workflow);
+    for( const filename of program.args ){
```

Comment:
This could result in partial pinning of files. It's not a big issue, but it would be better to do the validation up front, then process all files once we know that all input filenames exist.